corehq/apps/case_search/xpath_functions/ancestor_functions.py

```diff
@@ -23,22 +26,18 @@ def ancestor_comparison_query(context, node):
     """Return a query that will fulfill the filter on the related case.
 
     :param node: a node returned from eulxml.xpath.parse of the form `parent/grandparent/property = 'value'`
-
-    Since ES has no way of performing joins, we filter down in stages:
-    1. Find the ids of all cases where the condition is met
-    2. Walk down the case hierarchy, finding all related cases with the right identifier to the ids
-    found in (1).
-    3. Return the lowest of these ids as a related case query filter
     """
 
-    # fetch the ids of the highest level cases that match the case_property
-    # i.e. all the cases which have `property = 'value'`
-    case_ids = _parent_property_lookup(context, node)
-
     # extract ancestor path:
     # `parent/grandparent/property = 'value'` --> `parent/grandparent`
-    ancestor_path = node.left.left
-    return walk_ancestor_hierarchy(context, ancestor_path, case_ids)
+    ancestor_path = serialize(node.left.left)
+    parsed_ancestor_path = parse_xpath(ancestor_path)
+
+    case_property = serialize(node.left.right)
+    value = serialize(node.right)
+    parsed_ancestor_filter = parse_xpath(f'{case_property}{node.op}{value}')
```

Comment:
nit: You could just do this:
```python
ancestor_filter = BinaryExpression(case_property, node.op, value)
```

Reply:
Ah great, I couldn't figure how to do that [6f6fa88](https://github.com/dimagi/commcare-hq/pull/33014/commits/6f6fa88a64d8fe1bd4dc5a3483a89f1ecdf086a6)